helper.mac.fish

```diff
@@ -77,22 +81,98 @@ function findRequiredMinMacOS
 end
 
 function opensslVersion
-  set -l oversion $argv[1]
+  set -gx OPENSSL_VERSION $argv[1]
+end
 
-  if test "$oversion" = ""
-    set -e OPENSSL_VERSION
-    return 0
+function downloadOpenSSL
+  if test -z $OPENSSL_VERSION
+    echo "Need OPENSSL_VERSION global variable!"
+    return 1
   end
 
-  switch $oversion
-    case '1.0.2'
-      set -gx OPENSSL_VERSION $oversion
+  set -l directory $WORKDIR/work/openssl
+  set -l url https://www.openssl.org/source/openssl-$OPENSSL_VERSION.tar.gz  
+  mkdir -p $directory
+  cd $directory
+  echo "Downloading sources to $directory from URL: $url"
+  curl -LO $url
+  rm -rf openssl-$OPENSSL_VERSION
+  tar -xzvf openssl-$OPENSSL_VERSION.tar.gz
+end
 
-    case '1.1.1'
-      set -gx OPENSSL_VERSION $oversion
+function buildOpenSSL
+  if test "$OPENSSL_SOURCE_DIR" = ""
+    echo "OPENSSL_SOURCE_DIR is not defined. Please download source codes before building OpenSSL!"
+    return 1
+  else if ! test -d $OPENSSL_SOURCE_DIR
+      echo "Directory $OPENSSL_SOURCE_DIR does not exist!"
+      return 1
+  end
+  cd $OPENSSL_SOURCE_DIR
+  mkdir build
+  
+  if test -z "$ARCH"
+    echo "ARCH is not set! Can't decide wether to build OpenSSL for arm64 or x86_64."
```

Comment:
Why cannot not detect my System arch? And just build it?

Reply:
It's detected right at the beginning. It's just a matter of checking whether it's still good (the environment wasn't changed or we initialized oskar at some weird place).